### DISCUSSION TITLE
How the arg `holomorphic` affects the automatic differentiation?

### DISCUSSION THREAD

**Discussion by GinFungYJF** [jax-ml/jax#9533] [Q&A] [answered]
Many APIs of automatic differentiation has the arg `holomorphic`. In the source code, it will check whether the inputs and outputs  of func are complex when the `holomorphic` is `True`. But I can't see how it affects the result or process of automatic differentiation. Do I miss something? 
![image](https://user-images.githubusercontent.com/20573763/153546403-607c456a-9913-4d75-a79a-357871020a9d.png)

**Comment by YouJiacheng** [comment, marked as answer]
May I recommend the autodiff cook book: https://jax.readthedocs.io/en/latest/notebooks/autodiff_cookbook.html#complex-numbers-and-differentiation

**Reply by GinFungYJF** [reply]
I think I have got it from the autodiff cook book. Thank you so much.
